### PR TITLE
Reconcile CephObjectStores, CephObjectStoreUsers, and CephFilesystems

### PR DESCRIPTION
**Pull request by priyanka19-98**
Modified the handling of functions `ensureCephObjectStore()` , `ensureCephObjectStoreUsers()`, `ensureCephFileSystems()` by removing check for each function related Status and removing boolean from StorageClusterStatus struct.

Signed off by: Priyanka Jiandani <pjiandan@redhat.com>

### CODE REVIEW

**Comment by xenolinux** [comment]
> _No description provided._

Can you add description of this change here?

**Comment by priyanka19-98** [comment]
> > _No description provided._
> 
> Can you add description of this change here?

Done!

File: deploy/olm-catalog/ocs-operator/manifests/storagecluster.crd.yaml [hunk 1]
```diff
@@ -2386,12 +2326,6 @@ spec:
           properties:
             cephBlockPoolsCreated:
               type: boolean
-            cephFilesystemsCreated:
-              type: boolean
-            cephObjectStoreUsersCreated:
-              type: boolean
-            cephObjectStoresCreated:
-              type: boolean
```
Comment (umangachapagain) [inline deploy/olm-catalog/ocs-operator/manifests/storagecluster.crd.yaml:2394]: Except for these, all other changes in this file is wrong. 
Can you try generating again on top of latest master?

**Comment by priyanka19-98** [comment]
@jarrpa  please review.

**Comment by jarrpa** [comment]
> If we are removing most of these objects from Status, why are we still keeping `CephBlockPoolsCreated` ?
> Is there any reason for this to be treated specially?

We still want to allow the admin to delete CephBlockPools and associated StorageClasses if they so want.

**Pull request merged**